platform/darwin/src/MGLNetworkConfiguration.m

```diff
@@ -65,9 +65,13 @@ - (NSURLSessionConfiguration *)defaultSessionConfiguration {
 }
 
 - (void)startDownloadEvent:(NSString *)urlString type:(NSString *)resourceType {
-    if (urlString && ![self.events objectForKey:urlString]) {
-        [self.events setObject:@{ MGLStartTime: [NSDate date], MGLResourceType: resourceType } forKey:urlString];
+    NSString *retainedUrlString = urlString;
```

Comment:
I don't think this is sufficient. AFAIK `urlString` should be retained by this method  anyway - how about using `copy` here?

Reply:
I consider using copy but an increase in the retain count would do the job better.